code/parachain/frame/composable-support/src/types/rational.rs

```diff
@@ -0,0 +1,145 @@
+use core::num::NonZeroU64;
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::RuntimeDebug;
+use scale_info::TypeInfo;
+
+#[derive(
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	RuntimeDebug,
+	Encode,
+	Decode,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+/// Represents a [rational number], with 64 bit numbers for the numerator and denominator.
+///
+/// [rational number]: https://en.wikipedia.org/wiki/Rational_number
+pub struct Rational64 {
+	n: u64,
+	d: NonZeroU64,
+}
+
+impl Rational64 {
+	pub const fn new(n: u64, d: u64) -> Option<Self> {
+		match NonZeroU64::new(d) {
+			Some(d) => Some(Self { n, d }),
+			None => None,
+		}
+	}
+
+	pub const fn n(&self) -> u64 {
+		self.n
+	}
+
+	pub const fn d(&self) -> NonZeroU64 {
+		self.d
+	}
+}
+
+/// Used to construct a [`Rational64`] from literals. This will fail at compile time if the
+/// denominator is zero.
+///
+/// # Example
+///
+/// ```rust
+/// # use composable_support::{rational_64, types::rational::Rational64};
+/// const VALID: Rational64 = rational_64!(100 / 1);
+/// ```
+///
+/// ```rust,compile_fail
+/// # use composable_support::{rational_64, types::rational::Rational64};
+/// const INVALID: Rational64 = rational_64!(100 / 0);
+/// ```
+#[macro_export]
+macro_rules! rational_64 {
+	($n:literal / $d:literal) => {{
+		const RATIONAL64: Rational64 = match Rational64::new($n, $d) {
```

Comment:
nice.

Reply:
will macro work to init #[const] type SomePallet = rational!(41/0)  or like?

Reply:
i mean where is const get impl marco

Reply:
forgot to implement that - thanks for the reminder :+1:

Reply:
Implemented in https://github.com/ComposableFi/composable/commit/316007c947a581f670d7bee2279fc495a923dd01